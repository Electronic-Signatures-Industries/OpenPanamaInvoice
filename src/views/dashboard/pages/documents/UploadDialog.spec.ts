import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import UploadDialog from "./UploadDialog.vue";

describe("UploadDialog.vue", () => {
  const sentMock = jest.fn<void, [File]>();
  let wrapperComponent: Wrapper<Vue>;
  let component: Wrapper<UploadDialog>;

  beforeEach(() => {
    const localVue = createLocalVue();

    // to render vuetify dialog, vuetify requires the v-app component
    // so we pack our component into a "bridge" component
    const App = localVue.component("App", {
      components: { UploadDialog },
      data() {
        return { show: true };
      },
      methods: {
        result: sentMock,
      },
      template: `
        <v-app>
          <upload-dialog
            id="dialog"
            :loading="false"
            :show.sync="show"
            :uploadStatus="null"
            @result="result"
          />
        </v-app>
      `,
    });

    wrapperComponent = mount(App, {
      localVue,
      vuetify: new Vuetify(),
      attachToDocument: true,
    });

    component = wrapperComponent.find("#dialog");
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should hide when the 'Cancel' button is pressed", async () => {
    const button = component.find("[data-cy=cancel-button]");
    button.trigger("click");
    expect(wrapperComponent.vm.$data["show"]).toBeFalsy();
  });

  it("should sent the data when 'Confirm' button is pressed", async () => {
    const mockFile = new File(["Lorem Ipsum"], "test.txt");
    component.setData({ files: mockFile });
    const button = component.find("[data-cy=ok-button]");
    button.trigger("click");

    expect(sentMock).toBeCalledTimes(1);
    const sentFile = sentMock.mock.calls[sentMock.mock.calls.length - 1][0];
    expect(sentFile).toEqual(mockFile);
  });
});
