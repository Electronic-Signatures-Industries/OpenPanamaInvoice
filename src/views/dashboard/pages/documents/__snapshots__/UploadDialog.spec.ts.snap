// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDialog.vue should match snapshot 1`] = `<div role="dialog" class="v-dialog__container" id="dialog"></div>`;
